module/math/ndarray_cg/src/d2/mat3x3h/transformation.rs

```diff
@@ -144,3 +144,102 @@ where
   let dir = sub( &center, &eye );
   loot_to_rh( eye, dir, up )
 }
+
+/// Creates rotation matrix from consequtive rotation around X, Y and Z axes
+///
+/// # Parameters
+/// - `x`: The angle of rotation around X axis in radians.
+/// - `y`: The angle of rotation around Y axis in radians.
+/// - `z`: The angle of rotation around Z axis in radians.
+///
+/// # Returns
+/// - 4x4 rotation matrix.
+#[ inline ]
+pub fn rot< E > ( x : E, y : E, z : E ) -> Mat4< E >
+where
+  E : MatEl + nd::NdFloat,
+  Mat4< E > : RawSliceMut< Scalar = E >,
+{
+  // https://en.wikipedia.org/wiki/Rotation_matrix
+  let e11 = y.cos() * z.cos();
+  let e12 = x.sin() * y.sin() * z.cos() - x.cos() * z.sin();
+  let e13 = x.cos() * y.sin() * z.cos() + x.sin() * z.sin();
+
+  let e21 = y.cos() * z.sin();
+  let e22 = x.sin() * y.sin() * z.sin() + x.cos() * z.cos();
+  let e23 = x.cos() * y.sin() * z.sin() - x.sin() * z.cos();
+
+  let e31 = -( y.sin() );
+  let e32 = x.sin() * y.cos();
+  let e33 = x.cos() * y.cos();
```

Comment:
I think it will be better to store the result of (x,y,z).(sin, cos) in a variable and then reuse. rather than recalculating it over and over. But who knows, make the compiler is smart enough to optimize it

Reply:
Compiler probably will optimize it out. Consider [this](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DEArgoKkl9ZATwDKjdAGFUtEywZ7HAGTwNMAHLuAEaYxHoADqgKhHYMLm4ekdGxAr7%2BQSyh4VyWmNa2AkIETMQECe6euVaYNnHFpQTpgSFheuaNFUnVJWXNmdkSAJSWqCbEyOwcAKQATADM0wCsAEIMqAD6LIbA9MsAItMaAIIRJsEA1FQMF8SoBBAX6lfzs6QXAJ4gL28XAF7fKivIYXAC003m2AuyxWPwhsK4ADZoUt9tCAOwrI7HC64i70AgXTBcLjQ%2BZoj4AOjQCggIIAVP9qdE6fDsXj8ZhCcTZmS0apKTEGHSLoyqUKRYy/szaSDQU8ZZKmRKhmyThyCUSuPM%2BQqaUrxX4ldL9SC5rCBSrRcqjar5liTuy8ZrMLNSRCKYqGTbhXaHTjnVyibNeR6FVaxYLbdbpVbzXqWd6TYm1QHcS6FrqBabrYbfTGo/n5Zbo1Kvamnemg5h5u7yWDHnmRX7K5zua8s4WDeX7a2XfMdWHs4ncz3/bjWzDW7jibktb9ifN3hpKRpSNPg3PXQuFsvV%2Bv1RyibX3jWFwO92uNyu1xcb5f3lx95PUdjpujDo6TgB6b8XACSVCfGMFxmJg0KIho2y2tMkHvFgyAGMQ4GEBcTAKBBGhnMEsEaBcRAXEoKGEn4%2BEIOBYwEGcBAgNi2FXDc0G%2Bm%2BmKtuY6AgCACB%2BDRIDBAYyAANYbMEqCqBA76wncDwXLMlLvPM8kXJISl%2Bhi%2Bwto6H7YhwIy0JwSy8J4HBaKQqCcAASmYhIKGMEzgXMta8DRJm6SMgkgEsa76RwkhGZovDmRwvAKCAa4uVoIxwLASBoCwER0GE5CUHFCX0OExBcAAHLMa40LQBBhKFEDBAFpDBH4pRfNwvAVcwxAfAA8sE2i1C5PCkHFbCCI1DC0NVplYMEJjAE4Yi0KFNWkFg2xGOIrnTXgyF1AAbpgk2mZgqi1CYhWcB1PH5GVtB4MExBVS4WBlQQxB4Cw%2B28GtxCiUo%2ByYLNux%2BKArkjFQBjAAoABqeCYAA7o1ESMA9MiCCIYjsFIMPyEoahlbouQGEYICmOY%2BinaFkAjKgESFAwk2gk4FzEwQoL0GttAevMgVPbdWAE3SeQFHEDgMM4riVBI6KkD4fgtFkbRLNIUQxKTXRVEL0upAw/StOEkuc219S9OU/PdELNR1EU2sq%2BLavSB0ZRy4LljG6LAwS5IIy2eMkzDPoBn%2BQtQW3NZyAXE%2BOWrhcEC4IQJDQgsXBDM5AVDCM5FMFg4Qcx5Xnu75numUFIVhaQEVuensyZ4FnAxz9IxPTE9iSEAA%3D). I don't really understand assembly but it seems that compiler precomputes cos and sin and then just moves them around the registers calculating elements of matrix